client/opentracing/src/main/java/io/opentracing/APMSpan.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.opentracing;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.logging.Logger;
+
+import org.hawkular.apm.api.model.trace.CorrelationIdentifier;
+import org.hawkular.apm.api.model.trace.CorrelationIdentifier.Scope;
+import org.hawkular.apm.api.model.trace.NodeType;
+import org.hawkular.apm.client.api.reporter.TraceReporter;
+import org.hawkular.apm.client.opentracing.NodeBuilder;
+import org.hawkular.apm.client.opentracing.TraceContext;
+
+import io.opentracing.AbstractSpanBuilder.Reference;
+
+/**
+ * The APM span.
+ *
+ * @author gbrown
+ */
+public class APMSpan extends AbstractSpan {
+
+    private static final Logger log = Logger.getLogger(APMSpan.class.getName());
+
+    private TraceContext traceContext;
+
+    private NodeBuilder nodeBuilder;
+
+    private String interactionId;
+
+    /**
+     * @param builder The span builder
+     * @param reporter The trace reporter
+     */
+    public APMSpan(APMSpanBuilder builder, TraceReporter reporter) {
+        super(builder.operationName, builder.start);
+
+        init(builder, reporter);
+    }
+
+    protected void init(APMSpanBuilder builder, TraceReporter reporter) {
+        // Check references
+        for (Reference ref : builder.references) {
+            if (ref.getReferenceType() == References.CHILD_OF) {
```

Comment:
could you please refactor this to two separate methods? One for ChildOf and one for FollowsFrom?